converter/fluka/helper_parsers/scoring_parser.py

```diff
@@ -1,36 +1,148 @@
 from dataclasses import dataclass
+from typing import Optional, Union
 
 from converter.fluka.helper_parsers.detector_parser import Detector, parse_detector
 
+_particle_mappings: dict[str, str] = {
+    'Neutron': 'NEUTRON',
+    'Proton': 'PROTON',
+    'Pion π-': 'PION-',
+    'Pion π+': 'PION+',
+    'Anti-neutron': 'ANEUTRON',
+    'Anti-proton': 'APROTON',
+    'Kaon κ-': 'KAON-',
+    'Kaon κ+': 'KAON+',
+    'Kaon κ0': 'KAONZERO',
+    'Kaon κ~': 'AKAONZER',
+    'Muon µ-': 'MUON-',
+    'Muon µ+': 'MUON+',
+    'Deuteron': 'DEUTERON',
+    'Triton': 'TRITON',
+    'Helium-3': 'HELIUM3',
+    'Helium-4': 'HELIUM4',
+    'Heavy ions': 'HEAVYION',
+}
+
+__supported_filter_keywords = ('A', 'Z')
+
+
+@dataclass
+class CustomFilter:
+    """Class representing CustomFilter"""
+
+    name: str
+    a: int
+    z: int
+
+
+@dataclass
+class ParticleFilter:
+    """Class representing ParticleFilter"""
+
+    name: str
+    particle: str
+
+
+@dataclass
+class Quantity:
+    """Class representing Quantity"""
+
+    name: str
+    scoring_filter: Optional[Union[CustomFilter, ParticleFilter]]
+    modifiers: list[any]  # unused
+    keyword: str = 'DOSE'
+
 
 @dataclass
 class Scoring:
-    """Class representing Scoring"""
+    """Class representing Scorings for Fluka output"""
 
-    detectorUuid: str
+    quantities: list[Quantity]
     detector: Detector
-    name: str = ""
-    quantity: str = "DOSE"
+    output_unit: int = 21
+
+
+def get_particle_filter(filter_dict: dict) -> Optional[ParticleFilter]:
+    """Creates ParticleFilter from dictionary.
+
+    Returns None if filter cannot be created for Fluka.
+    """
+    particle = filter_dict['particle']
+    if particle.get('name') not in _particle_mappings:
+        return None
+
+    return ParticleFilter(name=particle['name'], particle=_particle_mappings[particle['name']])
+
+
+def get_custom_filter(filter_dict: dict) -> list[CustomFilter]:
+    """Creates CustomFilter from dictionary.
+
+    Returns None if filter cannot be created for Fluka.
+    """
+    if not filter_dict.get('rules'):
+        return None
```

Comment:
```suggestion
        return []
```

return empty list or adjust type hints for the output

Reply:
Type hint was misleading, function returns None or CustomFilter.
I forgot to update it after the refactoring.